yarn-project/sequencer-client/src/sequencer/public_processor.ts

```diff
@@ -105,6 +105,13 @@ export class PublicProcessor {
     const callStackItem = new PublicCallStackItem(contractAddress, txRequest.functionData, publicCircuitOutput);
     const publicCallStackPreimages: PublicCallStackItem[] = times(PUBLIC_CALL_STACK_LENGTH, PublicCallStackItem.empty);
 
+    // set the msgSender for each call in the call stack
+    for (let i = 0; i < publicCallStackPreimages.length; i++) {
+      const isDelegateCall = publicCallStackPreimages[i].publicInputs.callContext.isDelegateCall;
+      publicCallStackPreimages[i].publicInputs.callContext.msgSender = isDelegateCall
+        ? callStackItem.publicInputs.callContext.msgSender
+        : callStackItem.contractAddress;
+    }
```

Comment:
Shouldn't the circuit skip these checks for empty call stack items?

Reply:
Hmm, it currently doesn't, but it probably should. I wonder though if there should be a specific flag on the call stack item indicating if it is 'empty' rather than simply testing if 'xyz == 0' which seems rather arbitrary. What do you think @iAmMichaelConnor ?

Reply:
I do see what you mean although there is a school of thought that that dictates we should not modify our actual code to handle testing variants

Reply:
> we should not modify our actual code to handle testing variants

Fully agree. The problem here is that these are not testing variants: it's that these empty values are used in actual code to pad to the necessary number of items in the array of call stack items. Anyhow, I think @PhilWindle was tackling this in the circuit. so we should be able to remove this once the isEmpty check is added there.